src/util/bitvector.h

```diff
@@ -69,339 +68,198 @@ class CVC4_PUBLIC BitVector {
     return *this;
   }
 
-  unsigned getSize() const
-  {
-    return d_size;
-  }
+  /* Get size (bit-width). */
+  unsigned getSize() const;
+  /* Get value. */
+  const Integer& getValue() const;
 
-  const Integer& getValue() const
-  {
-    return d_value;
-  }
-
-  Integer toInteger() const
-  {
-    return d_value;
-  }
-
-  /**
-   * Return Integer corresponding to two's complement interpretation of bv.
-   */
-  Integer toSignedInteger() const
-  {
-    unsigned size = d_size;
-    Integer sign_bit = d_value.extractBitRange(1, size - 1);
-    Integer val = d_value.extractBitRange(size - 1, 0);
-    Integer res = Integer(-1) * sign_bit.multiplyByPow2(size - 1) + val;
-    return res;
-  }
-
-  size_t hash() const
-  {
-    return d_value.hash() + d_size;
-  }
-
-  std::string toString(unsigned int base = 2) const
-  {
-    std::string str = d_value.toString(base);
-    if (base == 2 && d_size > str.size())
-    {
-      std::string zeroes;
-      for (unsigned int i = 0; i < d_size - str.size(); ++i)
-      {
-        zeroes.append("0");
-      }
-      return zeroes + str;
-    }
-    else
-    {
-      return str;
-    }
-  }
-
-  BitVector setBit(uint32_t i) const
-  {
-    CheckArgument(i < d_size, i);
-    Integer res = d_value.setBit(i);
-    return BitVector(d_size, res);
-  }
+  /* Return value. */
+  Integer toInteger() const;
+  /* Return Integer corresponding to two's complement interpretation of this. */
+  Integer toSignedInteger() const;
+  /* Return (binary) string representation. */
+  std::string toString(unsigned int base = 2) const;
 
-  bool isBitSet(uint32_t i) const
-  {
-    CheckArgument(i < d_size, i);
-    return d_value.isBitSet(i);
-  }
+  /* Return hash value. */
+  size_t hash() const;
 
+  /* Set bit at index 'i'. */
+  BitVector setBit(uint32_t i) const;
+  /* Return true if bit at index 'i' is set. */
+  bool isBitSet(uint32_t i) const;
 
-  /**
-   * Return k if the integer is equal to 2^{k-1} and zero otherwise.
-   */
-  unsigned isPow2() {
-    return d_value.isPow2();
-  }
+  /* Return k if the value of this is equal to 2^{k-1}, and zero otherwise. */
+  unsigned isPow2();
```

Comment:
This could be declared as `unsigned isPow2() const;`.

Reply:
Done